vyper/codegen/function_definitions/external_function.py

```diff
@@ -86,10 +86,12 @@ def handler_for(calldata_kwargs, default_kwargs):
         # a sequence of statements to strictify kwargs into memory
         ret = ["seq"]
 
-        # ensure calldata is at least of minimum length
-        args_abi_t = calldata_args_t.abi_type
-        calldata_min_size = args_abi_t.min_size() + 4
-        ret.append(["assert", ["ge", "calldatasize", calldata_min_size]])
+        # run check if 0 selector; otherwise selector table checks equality
```

Comment:
interesting. will have to think about this more

Reply:
any further thoughts on this?

Reply:
i don't think this is right. this elides the check in all cases except when the selector is 0.

Reply:
are you saying that it should elide the check in all cases except when the selector has trailing zeroes?

"however when there are no trailing 0s, it can be omitted because the selector table already checks exact equality of the selector."

Reply:
nono, other way around

Reply:
decided to remove this change. assert calldatasize lt 4 is not redundant to this check; so even if selector is 0 we want to keep this check.